domains/client/domain-operator/src/domain_bundle_proposer.rs

```diff
@@ -204,6 +255,9 @@ where
                 estimated_bundle_weight = next_estimated_bundle_weight;
                 bundle_size = next_bundle_size;
                 extrinsics.push(pending_tx_data.clone());
+
+                self.previous_bundled_tx
```

Comment:
We seem to just add the bundle after submission but there is always a possiblity that bundle never made it into Consensus node TX pool. 
I would suggest checking the Consensus node TX pool if this bundle is included without any failures and then only proceed to index this here

Reply:
>but there is always a possiblity that bundle never made it into Consensus node TX pool.

Do you mean the local consensus node's tx pool? bundle should always be able to submit to the local consensus node tx pool, if it is failed then most likely there is a bug.

Reply:
The bundled tx hashes are cleared whenever the consensus chain tip is changed, which is because the bundle may not be included by the next consensus block due to:
- the bundle is dropped silently during propagation due to network issue
- the bundle can't fit into the next consensus block or didn't arrive at the block author's tx pool in time, and becomes invalid as the head ER is changed
- etc

We can't predict these situations locally thus we clear the bundled tx hashes whenever the consensus chain tip is changed to have more retry.

Reply:
While it is possible to monitor consensus tx pool and based on that we can make decision on whether to clear the cache or not. The implementation would be similar to how substrate transaction pool monitors block import/finalization, which can be very involved and in my opinion, not worth the effort.